web3/middleware/pythonic.py

```diff
@@ -128,7 +155,7 @@ def is_array_of_dicts(value):
     'gasUsed': to_integer_if_hex,
     'size': to_integer_if_hex,
     'timestamp': to_integer_if_hex,
-    'hash': to_ascii_if_bytes,
+    'hash': apply_formatter_if(to_hexbytes(32), is_not_null),
```

Comment:
this is surprisingly legible.  I was skeptical when I wrote it.  I think the argument order could be reversed at some point for better legibility

```python
apply_formatter_if(is_not_null, to_hexbytes(32))
```

Reply:
Yeah, it's grown on me.  I agree about swapping the arguments. When I was first getting familiar with `apply_formatter_if`, my expectation was that the test would come first. (it curries a bit more nicely, too, I think)